src/spider/tdl/parser/ast/node_impl/type_impl/container_impl/Map.hpp

```diff
@@ -0,0 +1,66 @@
+#ifndef SPIDER_TDL_PARSER_AST_NODE_IMPL_TYPE_IMPL_CONTAINER_IMPL_MAP_HPP
+#define SPIDER_TDL_PARSER_AST_NODE_IMPL_TYPE_IMPL_CONTAINER_IMPL_MAP_HPP
+
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <string>
+
+#include <ystdlib/error_handling/ErrorCode.hpp>
+#include <ystdlib/error_handling/Result.hpp>
+
+#include <spider/tdl/parser/ast/Node.hpp>
+#include <spider/tdl/parser/ast/node_impl/Type.hpp>
+#include <spider/tdl/parser/ast/node_impl/type_impl/Container.hpp>
+
+namespace spider::tdl::parser::ast::node_impl::type_impl::container_impl {
+class Map : public Container {
+public:
+    // Types
+    enum class ErrorCodeEnum : uint8_t {
+        UnsupportedKeyType = 1,
+    };
+
+    using ErrorCode = ystdlib::error_handling::ErrorCode<ErrorCodeEnum>;
+
+    // Factory function
+    /**
+     * @param key_type
+     * @param value_type
+     * @return A result containing a unique pointer to a new `Map` instance with the given key and
+     * value types on success, or an error code indicating the failure:
+     * - Map::ErrorCodeEnum::UnsupportedKeyType if the `key_type` is not supported.
+     * - Forwards `validate_child_node_type`'s return values.
```

Comment:
Switch the order since we first check `validate_child_node_type` then check whether the key type is supported.
```suggestion
     * - Forwards `validate_child_node_type`'s erroneous return values.
     * - Map::ErrorCodeEnum::UnsupportedKeyType if the `key_type` is not supported.
```

Reply:
As discussed offline, we put forward last.